src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.cs

```diff
@@ -286,10 +287,37 @@ public static void UnknownUrl_Throws_EndpointNotFoundException()
 
         EndpointNotFoundException exception = Assert.Throws<EndpointNotFoundException>(() =>
         {
-            using (ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(notFoundUrl)))
+            try
             {
-                IWcfService serviceProxy = factory.CreateChannel();
-                string response = serviceProxy.Echo("Hello");
+                using (
+                    ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding,
+                        new EndpointAddress(notFoundUrl)))
+                {
+                    IWcfService serviceProxy = factory.CreateChannel();
+                    string response = serviceProxy.Echo("Hello");
+                }
+            }
+            catch (EndpointNotFoundException)
+            {
+                throw;
+            }
+            catch (CommunicationException ce)
+            {
+                if (ce.InnerException == null)
+                    throw;
+
+                if (ce.InnerException.GetType() == typeof (HttpRequestException))
+                {
+                    var httpReqExcep = ce.InnerException as HttpRequestException;
+                    StringBuilder sb = new StringBuilder();
+                    sb.Append("Received HttpRequestException with unknown error code ")
+                        .Append(ce.InnerException.HResult).AppendLine();
```

Comment:
Nit: formatting a bit odd.  It might look better with something like:
   sb.Append(...)
       .Append(...)
       .Append(...)
Don't block on this suggestion.